### ISSUE TITLE
Add widget

### ISSUE THREAD

**Issue by MXC48** [Akylas/oss-weather#37]
it would be nice to be able to add a widget with the time, date and of course the weather
a bit like can do géometric weather
![Screenshot_20230429-215612_Trebuchet.png](https://user-images.githubusercontent.com/123580315/235321787-7d8a6718-40bf-4153-b683-8b352e2c8c48.png)

**Comment by farfromrefug** [comment]
@GuiSousa135 @MXC48 sorry but I am really not sure I will implement something like this. My app always was meant to use as less battery as possible.  I am really not sure this is something needed

**Comment by Subhashis2007** [comment]
> @GuiSousa135 @MXC48 sorry but I am really not sure I will implement something like this. My app always was meant to use as less battery as possible. I am really not sure this is something needed

can you rethink of it a little, a toggle to enable a widget, with looks same as of google's at a glance, bcz most of the custom rom users either use Pixel Launcher or Lawnchair with Googleapp to get weather at a glance on homescreen, there is also another type of weather implementation in launcher3 based roms, which works without location permissions and setting a custom location.

**Comment by farfromrefug** [comment]
@Subhashis2007 i can think about it sure! Do you have screenshots of what you would prefer?

>  there is also another type of weather implementation in launcher3 based roms, which works without location permissions and setting a custom location.
Do you have reference? code or anything i could look at?

**Comment by Subhashis2007** [comment]
> @Subhashis2007 i can think about it sure! Do you have screenshots of what you would prefer?
> 
> > there is also another type of weather implementation in launcher3 based roms, which works without location permissions and setting a custom location.
> > Do you have reference? code or anything i could look at?

![IMG_20240214_211903_981](https://github.com/Akylas/oss-weather/assets/113865986/ac09a70c-f70c-4b5b-a5d2-e5086afce1f9)

See the circled portion

**Comment by farfromrefug** [comment]
@Subhashis2007 so what you are looking for is a simple widget to show current weather temperature/icon ?

**Comment by Subhashis2007** [comment]
> @Subhashis2007 so what you are looking for is a simple widget to show current weather temperature/icon ?
yeahh

**Comment by farfromrefug** [comment]
@Subhashis2007 sorry for all the questions! i dont use widgets at all (even if i am on LOS and using custom launcher ;)). And you expect the widget to report the weather of the current weather location in OSS Weather?

**Comment by Subhashis2007** [comment]
> @Subhashis2007 sorry for all the questions! i dont use widgets at all (even if i am on LOS and using custom launcher ;)). And you expect the widget to report the weather of the current weather location in OSS Weather?

yep, I know a lot of people don't use weather widgets but it seems modern,

the rest is up to you

**Comment by farfromrefug** [comment]
@Subhashis2007 i ll think about it!

**Comment by drp4positive** [comment]
I would be happy to have widget options  as described by @GuiSousa135 in 2x2, 4x2 and 1x1 (as by @ghost). Then I could dump my other weather app which I only use for widgets for a small glance at the weather forecast for the next hours and not for details information.

**Comment by alahmed** [comment]
Widget support will be highly appreciated. Thanks

**Comment by farfromrefug** [comment]
@drp4positive @alahmed @GuiSousa135 @MXC48 I am really happy to announce that widgets have finally be added on android in latest release!
Please test it and report any issue/improvement you see

**Comment by farfromrefug** [comment]
@drp4positive all should be fixed now

**Comment by drp4positive** [comment]
Thanks @farfromrefug . It´s a thrill ;-)
Will start a yearly donation for you. I am enjoying your work and that has to be honoured.
